bionmr_utils/md/process/select.py

```diff
@@ -19,67 +18,74 @@ def get_methyl_vectors(frame: Frame) -> List[Tuple[Atom, Atom]]:
                 ('ILE', 'CG2'): ['CG2', 'HG21'],
                 ('MET', 'CE1'): ['CE1', 'HE1']}
 
-    atom_pairs = []
+    C_ids = []
+    H_ids = []
 
-    for r in frame.asResidues:
-        for atom in r.asAtoms:
-            if (r.name.str, atom.name.str) in CH3_dict.keys():
-                C = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][0])]
-                H = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][1])]
-                atom_pairs.append((C, H))
+    for r in frame.residues:
+        for atom in r.atoms:
+            if (r.name, atom.name) in CH3_dict.keys():
+                C_ids.append(r[CH3_dict[(r.name, atom.name)][0]].id)
+                H_ids.append(r[CH3_dict[(r.name, atom.name)][1]].id)
 
-    return atom_pairs
+    C_atoms = frame.atoms.filter(aId.is_in(set(C_ids)))
+    H_atoms = frame.atoms.filter(aId.is_in(set(H_ids)))
+    atom_pairs_selection = (C_atoms, H_atoms)
 
+    return atom_pairs_selection

```

Comment:
Feels like already wrote that once :thinking: 

```suggestion
    C_atoms = []
    H_atoms = []

    for r in frame.residues:
        for atom in r.atoms:
            if (r.name, atom.name) in CH3_dict.keys():
                c_name, h_name = CH3_dict[r.name, atom.name]
                C_atoms.append(r[c_name])
                H_atoms.append(r[h_name])

    return AtomSelection(C_atoms), AtomSelection(H_atoms)
```